server/game/cards/08.1-TAK/MagTheMighty.js

```diff
@@ -1,25 +1,25 @@
 const DrawCard = require('../../drawcard.js');
-
 class MagTheMighty extends DrawCard {
     setupCardAbilities() {
         this.forcedReaction({
             when: {
                 afterChallenge: event => event.challenge.winner === this.controller && this.isParticipating()
             },
             target: {
-                cardCondition: card => card.location === 'play area' && card.controller === this.controller && card.getType() === 'character'
+                cardCondition: card => card.location === 'play area' && card.controller === this.controller && card.getType() === 'character' && card.canBeKilled()
             },
             handler: context => {
                 this.game.killCharacter(context.target);
                 this.game.addMessage('{0} is forced by {1} to kill {2}', context.player, this, context.target);
 
-                //TODO Technically should only trigger when the first kill was not saved
-                this.game.promptForSelect(context.event.challenge.loser, {
-                    activePromptTitle: 'Select a character (only when first kill was not saved)',
-                    cardCondition: card => card.location === 'play area' && card.controller === context.event.challenge.loser && card.getType() === 'character',
-                    source: this,
-                    onSelect: (player, card) => this.onCardSelected(player, card)
-                });
+                if(context.target.location && context.target.location === 'dead pile') {
```

Comment:
If we're going to fix Mag, a proper `then` mechanism has been introduced that should solve this issue. We should do that rather than continue this psuedo fix (which I don't think will work due to the way interrupts + events are processed). Maybe drop this from the PR and then PR separately?

Reply:
Will cut this from the PR and open up a seperate one for good old mag